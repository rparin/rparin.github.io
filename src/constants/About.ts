const AboutGen = {
  Initials: "RP",
  First: "Ralph",
  Last: "Parin",
  email: "r.parin070@gmail.com",
  titles: "Software Engineer, Web Developer, Gamer",
  sent: "I enjoy",
};

const AboutSent = [
  "reading light novels",
  "playing games",
  "watching movies",
  "learning new things",
];

const Socials = {
  github: "https://github.com/rparin",
  linkedIn: "https://www.linkedin.com/in/rparin",
};

const AboutMe = {
  Intro: "A little about me: my hobbies, interests, and goals.",

  Goals:
    "The thrill of crafting with my hands led me to computer science, where I aspire to make a meaningful difference in the lives of others, whether through a useful digital tool or a fun game.",
  GoalsSm:
    "Crafting drove me to computer science to make a difference, via games or digital tools.",

  Hobbies:
    "I cherish stories for the unique lessons they offer, particularly the no-happy-ending tales by my favorite author, Sugaru Miaki. They've taught me to embrace failure as a path to improvement.",
  HobbiesSm:
    "I enjoy reading for the lessons they offer. They've, taught me to overcome my regrets.",

  Interests:
    "One of my passion lies in creating memorable moments with friendsâ€”whether it's trying new flavors at local eateries, embarking on food adventures, or exploring exciting new places together.",
  InterestsSm:
    "One hobby of mine includes trying new foods and sightseeing with friends.",

  Gaming:
    "Gaming with my friends has become my digital backyard, as life takes us on different paths and distances grow, this space keeps us connected.",
  GamingSm:
    "Gaming with friends is my digital backyard, keeping me close despite life's changes.",
};

export { AboutGen, AboutSent, Socials, AboutMe };
