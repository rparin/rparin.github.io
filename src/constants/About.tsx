const AboutGen = {
  First: "Ralph",
  Last: "Parin",
  email: "r.parin070@gmail.com",
  titles: "Software Engineer, WebDeveloper, Gamer",
  sent: "I enjoy",
};

const AboutSent = [
  "reading light novels.",
  "playing games.",
  "watching movies.",
  "trying new things.",
];

export { AboutGen, AboutSent };
