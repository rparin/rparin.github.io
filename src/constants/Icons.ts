import { StaticImageData } from "next/image";
import git from "@icons/Github.svg";
import linked from "@icons/LinkedIn.svg";
import gmail from "@icons/GMail.svg";
import cpp from "@icons/Cpp.svg";
import css from "@icons/CSS.svg";
import js from "@icons/JS.svg";
import html from "@icons/HTML.svg";
import java from "@icons/Java.svg";
import ts from "@icons/TS.svg";
import angular from "@icons/Angular.svg";
import react from "@icons/React.svg";
import nodejs from "@icons/NodeJs.svg";
import fbase from "@icons/Firebase.svg";
import python from "@icons/Python.svg";
import ionic from "@icons/Ionic.svg";
import tailwind from "@icons/Tailwind.svg";
import pygame from "@icons/pygame.svg";
import sfml from "@icons/SFML.svg";
import openai from "@icons/OpenAi.svg";
import pandas from "@icons/Pandas.svg";
import flask from "@icons/Flask.svg";
import tensflow from "@icons/Tensorflow.svg";
import mapbox from "@icons/mapbox.svg";
import nextjs from "@icons/nextjs.svg";
import reactquery from "@icons/reactquery.svg";
import zod from "@icons/Zod.svg";
import springboot from "@icons/SpringBoot.svg";
import docker from "@icons/Docker.svg";

type Icon = {
  name: string;
  img: StaticImageData;
  alt: string;
  invert?: boolean;
  url?: string;
};

const SocialIcons = {
  github: {
    name: "Github",
    img: git,
    alt: "logo of Github",
  },
  linkedin: {
    name: "LinkedIn",
    img: linked,
    alt: "logo of LinkedIn",
  },
  gmail: {
    name: "Gmail",
    img: gmail,
    alt: "logo of Gmail",
  },
};

const DevTools = {
  angular: {
    name: "Angular",
    img: angular,
    alt: "logo of angular",
    url: "https://angular.io/",
  },
  react: {
    name: "React",
    img: react,
    alt: "logo of react",
    url: "https://react.dev/",
  },
  nextjs: {
    name: "Nextjs",
    img: nextjs,
    alt: "logo of nextjs",
    url: "https://nextjs.org/",
  },
  springboot: {
    name: "Spring Boot",
    img: springboot,
    alt: "logo of Spring Boot",
    url: "https://spring.io/projects/spring-boot",
  },
  reactquery: {
    name: "Tanstack Query",
    img: reactquery,
    alt: "logo of reactquery",
    url: "https://tanstack.com/query/latest",
  },
  zod: {
    name: "Zod",
    img: zod,
    alt: "logo of zod",
    url: "https://zod.dev",
  },
  docker: {
    name: "Docker",
    img: docker,
    alt: "logo of Docker",
    url: "https://www.docker.com",
  },
  mapbox: {
    name: "Mapbox",
    img: mapbox,
    alt: "logo of mapbox",
    url: "https://www.mapbox.com/",
  },
  ionic: {
    name: "Ionic",
    img: ionic,
    alt: "logo of ionic",
    url: "https://ionicframework.com/",
  },
  tailwind: {
    name: "Tailwind",
    img: tailwind,
    alt: "logo of tailwind",
    url: "https://tailwindcss.com/",
  },
  fbase: {
    name: "Firebase",
    img: fbase,
    alt: "logo of firebase",
    url: "https://firebase.google.com/",
  },
  nodejs: {
    name: "Nodejs",
    img: nodejs,
    alt: "logo of nodejs",
    url: "https://nodejs.org/en",
  },
  cpp: {
    name: "C++",
    img: cpp,
    alt: "logo of c++",
    url: "https://isocpp.org/",
  },
  css: {
    name: "CSS",
    img: css,
    alt: "logo of css",
    url: "https://developer.mozilla.org/en-US/docs/Web/CSS",
  },
  js: {
    name: "Javascript",
    img: js,
    alt: "logo of javascript",
    url: "https://developer.mozilla.org/en-US/docs/Web/javascript",
  },
  html: {
    name: "HTML",
    img: html,
    alt: "logo of html",
    url: "https://developer.mozilla.org/en-US/docs/Web/HTML",
  },
  ts: {
    name: "Typescript",
    img: ts,
    alt: "logo of typescript",
    url: "https://www.typescriptlang.org/",
  },
  python: {
    name: "Python",
    img: python,
    alt: "logo of python programming language",
    url: "https://www.python.org/",
  },
  java: {
    name: "Java",
    img: java,
    alt: "logo of java",
    url: "https://www.java.com/en/download/help/whatis_java.html",
  },
  pygame: {
    name: "Pygame",
    img: pygame,
    alt: "logo of pygame library",
    url: "https://www.pygame.org/",
  },
  sfml: {
    name: "SFML",
    img: sfml,
    alt: "logo of SFML library",
    url: "https://www.sfml-dev.org/",
  },
  tensflow: {
    name: "Tensorflow",
    img: tensflow,
    alt: "logo of tensorflow",
    url: "https://www.tensorflow.org/",
  },
  pandas: {
    name: "Pandas",
    img: pandas,
    alt: "logo of pandas library",
    url: "https://pandas.pydata.org/",
  },
  flask: {
    name: "Flask",
    img: flask,
    alt: "logo of flask framework",
    url: "https://flask.palletsprojects.com/en/3.0.x/",
  },
  openai: {
    name: "OpenAi",
    img: openai,
    alt: "logo of openai",
    url: "https://openai.com/",
  },
};

export { DevTools, SocialIcons };
export type { Icon };
