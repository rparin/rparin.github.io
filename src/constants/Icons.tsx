import { StaticImageData } from "next/image";
import git from "@icons/Github.svg";
import linked from "@icons/LinkedIn.svg";
import gmail from "@icons/GMail.svg";
import download from "@icons/Download.svg";
import cpp from "@icons/Cpp.svg";
import css from "@icons/CSS.svg";
import js from "@icons/JS.svg";
import html from "@icons/HTML.svg";
import java from "@icons/Java.svg";
import ts from "@icons/TS.svg";
import angular from "@icons/Angular.svg";
import react from "@icons/React.svg";
import nodejs from "@icons/NodeJs.svg";
import fbase from "@icons/Firebase.svg";
import python from "@icons/Python.svg";
import ionic from "@icons/Ionic.svg";
import tailwind from "@icons/Tailwind.svg";
import pygame from "@icons/pygame.svg";
import sfml from "@icons/SFML.svg";
import openai from "@icons/OpenAi.svg";
import pandas from "@icons/Pandas.svg";
import flask from "@icons/Flask.svg";
import tensflow from "@icons/Tensorflow.svg";
import mapbox from "@icons/mapbox.svg";
import nextjs from "@icons/nextjs.svg";
import reactquery from "@icons/reactquery.svg";

type Img = {
  img: StaticImageData;
  alt: string;
  invert?: boolean;
};

const MiscIcons = {
  download: {
    img: download,
    alt: "download icon",
  },
};

const SocialIcons = {
  github: {
    img: git,
    alt: "logo of Github",
  },
  linkedin: {
    img: linked,
    alt: "logo of LinkedIn",
  },
  gmail: {
    img: gmail,
    alt: "logo of Gmail",
  },
};

const CodeIcons = {
  angular: {
    img: angular,
    alt: "logo of angular",
  },
  react: {
    img: react,
    alt: "logo of react",
  },
  nextjs: {
    img: nextjs,
    alt: "logo of nextjs",
  },
  reactquery: {
    img: reactquery,
    alt: "logo of reactquery",
  },
  mapbox: {
    img: mapbox,
    alt: "logo of mapbox",
  },
  ionic: {
    img: ionic,
    alt: "logo of ionic",
  },
  tailwind: {
    img: tailwind,
    alt: "logo of tailwind",
  },
  fbase: {
    img: fbase,
    alt: "logo of firebase",
  },
  nodejs: {
    img: nodejs,
    alt: "logo of nodejs",
  },
  cpp: {
    img: cpp,
    alt: "logo of c++",
  },
  css: {
    img: css,
    alt: "logo of css",
  },
  js: {
    img: js,
    alt: "logo of javascript",
  },
  html: {
    img: html,
    alt: "logo of html",
  },
  ts: {
    img: ts,
    alt: "logo of typescript",
  },
  python: {
    img: python,
    alt: "logo of python programming language",
  },
  java: {
    img: java,
    alt: "logo of java",
  },
  pygame: {
    img: pygame,
    alt: "logo of pygame library",
  },
  sfml: {
    img: sfml,
    alt: "logo of SFML library",
  },
  tensflow: {
    img: tensflow,
    alt: "logo of tensorflow",
  },
  pandas: {
    img: pandas,
    alt: "logo of pandas library",
  },
  flask: {
    img: flask,
    alt: "logo of flask framework",
  },
  openai: {
    img: openai,
    alt: "logo of openai",
    invert: true,
  },
};

export { MiscIcons, CodeIcons, SocialIcons };
export type { Img };
