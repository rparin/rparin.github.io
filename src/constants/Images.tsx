import landBgR from "../assets/Landing/Red.svg";
import landBgP from "../assets/Landing/Purple.svg";
import landCode from "../assets/Landing/CodePy.webp";
import AboutCode from "../assets/About/CodeE.webp";
import AboutFood from "../assets/About/FoodE.webp";
import AboutGame from "../assets/About/LaptopE.webp";
import AboutBook from "../assets/About/BookE.webp";

const HomeImages = {
  bgR: {
    img: landBgR,
    alt: "Landing page background",
  },
  bgP: {
    img: landBgP,
    alt: "Landing page background",
  },
  code: {
    img: landCode,
    alt: "Landing page code background overlay",
  },
};

const AboutImages = {
  code: {
    img: AboutCode,
    alt: "Background image of study room",
  },
  book: {
    img: AboutBook,
    alt: "Background image of book in forest",
  },
  food: {
    img: AboutFood,
    alt: "Background image of food in cafe",
  },
  game: {
    img: AboutGame,
    alt: "Background image of laptop in urban city",
  },
};

export { HomeImages, AboutImages };
