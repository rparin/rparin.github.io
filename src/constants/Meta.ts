const og = {
  type: "website",
  url: "https://rparin.github.io/",
  title: "Ralph Parin | Portfolio",
  description:
    "A website showcasing some projects I've worked on and things about me. The technologies used to develop this website include React, Next.js, Tailwind CSS, and Shadcn-ui.",
  siteName: "rparin.github.io",
  images: [
    {
      url: "https://raw.githubusercontent.com/rparin/rparin.github.io/main/Banner.png",
    },
  ],
};

const Metadata = {
  metadataBase: new URL("https://rparin.github.io"),
  title: og.title,
  description: og.description,
  openGraph: og,
  other: {
    "google-site-verification": "hcmUVD9Z980Mb3ueRfyryxsObeKlNgrYfY7br2Eaenk",
  },
};

export { Metadata };
