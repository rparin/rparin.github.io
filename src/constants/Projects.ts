import { Icon } from "./Icons";
import { DevTools as dt } from "./Icons";

type pType = {
  name: string;
  img: string;
  alt: string;
  span?: boolean;
  github?: string;
  website?: string;
  icons: Array<Icon>;
  demo?: {
    img: string;
    alt: string;
  };
};

const Projects = {
  SArtNet: {
    name: "Spotify Artist Network",
    img: "https://raw.githubusercontent.com/rparin/SpotifyArtistNetwork/main/_preview/SpotifyArtistNetwork.webp",
    alt: "Spotify Artist Network Landing page screenshot",
    span: true,
    github: "https://github.com/rparin/SpotifyArtistNetwork",
    website: "https://spotify-artist-network.vercel.app",
    icons: [
      dt.react,
      dt.nodejs,
      dt.nextjs,
      dt.zod,
      dt.reactquery,
      dt.tailwind,
      dt.ts,
      dt.html,
    ],
    demo: {
      img: "https://raw.githubusercontent.com/rparin/SpotifyArtistNetwork/main/_preview/Demo.gif",
      alt: "Demo of Spotify Artist Network",
    },
  },
  CTMap: {
    name: "Clinical Trials Map",
    img: "https://raw.githubusercontent.com/rparin/CTMap/main/_preview/CTMap.webp",
    alt: "Clinical Trials Map App Screenshot",
    span: true,
    github: "https://github.com/rparin/CTMap",
    icons: [dt.react, dt.nodejs, dt.tailwind, dt.mapbox, dt.ts, dt.html],
    demo: {
      img: "https://raw.githubusercontent.com/rparin/CTMap/main/_preview/Demo.gif",
      alt: "Demo of Clinical Trials Map",
    },
  },
  STracker: {
    name: "Sleep Tracker",
    img: "https://raw.githubusercontent.com/rparin/Sleep-Tracker/main/preview/Sleep%20Tracker.webp",
    alt: "Sleep Tracker App Screenshot",
    span: true,
    github: "https://github.com/rparin/Sleep-Tracker",
    icons: [dt.angular, dt.fbase, dt.ionic, dt.html, dt.css, dt.ts],
    demo: {
      img: "https://raw.githubusercontent.com/rparin/Sleep-Tracker/main/preview/Demo.gif",
      alt: "Demo of Sleep Tracker App",
    },
  },
  Sudoku: {
    name: "Sudoku",
    img: "https://raw.githubusercontent.com/rparin/Sudoku/main/preview/Sudoku.webp",
    alt: "Sudoku Help Menu Screenshot",
    github: "https://github.com/rparin/Sudoku",
    icons: [dt.python, dt.pygame],
    demo: {
      img: "https://raw.githubusercontent.com/rparin/Sudoku/main/preview/Demo.gif",
      alt: "Demo of Sudoku game",
    },
  },
  PPSim: {
    name: "Prey Predator Sim",
    img: "https://raw.githubusercontent.com/rparin/Prey-and-Predator/main/preview/PreyPredSim.webp",
    alt: "Prey Predator Simulation Screenshot",
    github: "https://github.com/rparin/Prey-and-Predator",
    icons: [dt.cpp, dt.sfml],
    demo: {
      img: "https://raw.githubusercontent.com/rparin/Prey-and-Predator/main/preview/Demo.gif",
      alt: "Demo of Prey and Predator Sim",
    },
  },
  FViewer: {
    name: "Font Viewer",
    img: "https://raw.githubusercontent.com/rparin/Font-Viewer/main/preview/FViewer.webp",
    alt: "Font Viewer App Screenshot",
    span: true,
    github: "https://github.com/rparin/Font-Viewer",
    icons: [dt.angular, dt.fbase, dt.tensflow, dt.html, dt.css, dt.ts],
    demo: {
      img: "https://raw.githubusercontent.com/rparin/Font-Viewer/main/preview/demo.gif",
      alt: "Demo of Font Viewer Application",
    },
  },
  GCalc: {
    name: "Graphing Calculator",
    img: "https://raw.githubusercontent.com/rparin/Graphing-Calculator/main/preview/GCalc.webp",
    alt: "Graphing Calculator App Screenshot",
    span: true,
    github: "https://github.com/rparin/Graphing-Calculator",
    icons: [dt.cpp, dt.sfml],
    demo: {
      img: "https://raw.githubusercontent.com/rparin/Graphing-Calculator/main/preview/Demo.gif",
      alt: "Demo of Graphing Calc App",
    },
  },
  TEditor: {
    name: "Text Editor",
    img: "https://raw.githubusercontent.com/rparin/JText-Editor/main/preview/TextEditor.webp",
    alt: "Text Editor Screenshot",
    github: "https://github.com/rparin/JText-Editor",
    icons: [dt.java],
    demo: {
      img: "https://raw.githubusercontent.com/rparin/JText-Editor/main/preview/Demo.gif",
      alt: "Demo of Text Editor App",
    },
  },
  SSearch: {
    name: "Search Engine",
    img: "https://raw.githubusercontent.com/rparin/rparin.github.io/main/src/assets/Projects/SSearch/Search.webp",
    alt: "Search App Screenshot",
    span: true,
    icons: [dt.python, dt.openai, dt.flask, dt.pandas, dt.html, dt.css, dt.js],
    demo: {
      img: "https://raw.githubusercontent.com/rparin/rparin.github.io/main/src/assets/Projects/SSearch/Demo.gif",
      alt: "Demo of search engine",
    },
  },
};

// Items will be displayed based on array order
const ProjectItems: Array<pType> = [
  Projects.SArtNet,
  Projects.CTMap,
  Projects.Sudoku,
  Projects.PPSim,
  Projects.SSearch,
  Projects.STracker,
  Projects.GCalc,
  Projects.FViewer,
  Projects.TEditor,
];

export { Projects, ProjectItems };
export type { pType };
