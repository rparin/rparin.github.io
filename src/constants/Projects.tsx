import { StaticImageData } from "next/image";
import { Img } from "./Icons";
import { CodeIcons as ci } from "./Icons";

import SArtNet from "@project/SpotifyArtistNetwork.webp";
import CTMap from "@project/CTMap.webp";
import FViewer from "@project/FViewer.webp";
import GCalc from "@project/GCalc.webp";
import PPSim from "@project/PreyPredSim.webp";
import SSearch from "@project/Search.webp";
import STracker from "@project/Sleep Tracker.webp";
import Sudoku from "@project/Sudoku.webp";
import TEditor from "@project/TextEditor.webp";

import SArtNetDemo from "@demo/SpotifyArtistNetwork.gif";
import SSdemo from "@demo/Search.gif";
import CTDemo from "@demo/CTMap.gif";
import SudokuDemo from "@demo/Sudoku.gif";
import PPSimDemo from "@demo/PPSim.gif";
import GCalcDemo from "@demo/GCalc.gif";
import TEditorDemo from "@demo/TEditor.gif";
import FViewerDemo from "@demo/FViewer.gif";
import STDemo from "@demo/STracker.gif";

type pType = {
  name: string;
  img: StaticImageData;
  alt: string;
  span?: boolean;
  url?: string;
  website?: string;
  icons: Array<Img>;
  demo?: Img;
};

const Projects = {
  SArtNet: {
    name: "Spotify Artist Network",
    img: SArtNet,
    alt: "Spotify Artist Network Landing page screenshot",
    span: true,
    url: "https://github.com/rparin/SpotifyArtistNetwork",
    website: "https://spotify-artist-network.vercel.app",
    icons: [
      ci.react,
      ci.nodejs,
      ci.nextjs,
      ci.reactquery,
      ci.tailwind,
      ci.ts,
      ci.html,
    ],
    demo: {
      img: SArtNetDemo,
      alt: "Demo of Spotify Artist Network",
    },
  },
  CTMap: {
    name: "Clinical Trials Map",
    img: CTMap,
    alt: "Clinical Trials Map App Screenshot",
    span: true,
    url: "https://github.com/rparin/CTMap",
    icons: [ci.react, ci.nodejs, ci.tailwind, ci.mapbox, ci.ts, ci.html],
    demo: {
      img: CTDemo,
      alt: "Demo of Clinical Trials Map",
    },
  },
  STracker: {
    name: "Sleep Tracker",
    img: STracker,
    alt: "Sleep Tracker App Screenshot",
    span: true,
    url: "https://github.com/rparin/Sleep-Tracker",
    icons: [ci.angular, ci.fbase, ci.ionic, ci.html, ci.css, ci.ts],
    demo: {
      img: STDemo,
      alt: "Demo of Sleep Tracker App",
    },
  },
  Sudoku: {
    name: "Sudoku",
    img: Sudoku,
    alt: "Sudoku Help Menu Screenshot",
    url: "https://github.com/rparin/Sudoku",
    icons: [ci.python, ci.pygame],
    demo: {
      img: SudokuDemo,
      alt: "Demo of Sudoku game",
    },
  },
  PPSim: {
    name: "Prey Predator Sim",
    img: PPSim,
    alt: "Prey Predator Simulation Screenshot",
    url: "https://github.com/rparin/Prey-and-Predator",
    icons: [ci.cpp, ci.sfml],
    demo: {
      img: PPSimDemo,
      alt: "Demo of Prey and Predator Sim",
    },
  },
  FViewer: {
    name: "Font Viewer",
    img: FViewer,
    alt: "Font Viewer App Screenshot",
    span: true,
    url: "https://github.com/rparin/Font-Viewer",
    icons: [ci.angular, ci.fbase, ci.tensflow, ci.html, ci.css, ci.ts],
    demo: {
      img: FViewerDemo,
      alt: "Demo of Font Viewer Application",
    },
  },
  GCalc: {
    name: "Graphing Calculator",
    img: GCalc,
    alt: "Graphing Calculator App Screenshot",
    span: true,
    url: "https://github.com/rparin/Graphing-Calculator",
    icons: [ci.cpp, ci.sfml],
    demo: {
      img: GCalcDemo,
      alt: "Demo of Graphing Calc App",
    },
  },
  TEditor: {
    name: "Text Editor",
    img: TEditor,
    alt: "Text Editor Screenshot",
    url: "https://github.com/rparin/JText-Editor",
    icons: [ci.java],
    demo: {
      img: TEditorDemo,
      alt: "Demo of Text Editor App",
    },
  },
  SSearch: {
    name: "Search Engine",
    img: SSearch,
    alt: "Search App Screenshot",
    span: true,
    url: "https://github.com/rparin/StellerSearch",
    icons: [ci.python, ci.openai, ci.flask, ci.pandas, ci.html, ci.css, ci.js],
    demo: {
      img: SSdemo,
      alt: "Demo of search engine",
    },
  },
};

// Items will be displayed based on array order
const ProjectItems: Array<pType> = [
  Projects.SArtNet,
  Projects.CTMap,
  Projects.Sudoku,
  Projects.PPSim,
  Projects.SSearch,
  Projects.STracker,
  Projects.GCalc,
  Projects.FViewer,
  Projects.TEditor,
];

export { Projects, ProjectItems };
export type { pType };
