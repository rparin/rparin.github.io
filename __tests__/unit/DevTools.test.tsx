import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import { Projects } from "@/constants/Projects";
import DevTools from "@/components/ui/DevTools";

describe("testing DevTools Component Accessibility", () => {
  expect.extend(toHaveNoViolations);
  test("it should have no accessibility violations", async () => {
    const { container } = render(
      <div role="grid">
        <DevTools tools={Projects.HelloDog.icons} />
      </div>
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

test("it renders component unchanged", () => {
  const { container } = render(<DevTools tools={Projects.HelloDog.icons} />);
  expect(container).toMatchSnapshot();
});
