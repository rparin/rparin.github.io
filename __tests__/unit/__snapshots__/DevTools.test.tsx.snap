// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <div
    aria-label="Dev Tools"
    class="p-scroll row-span-3 mx-1 mt-1 flex flex-col items-center justify-start gap-2 overflow-y-auto md:row-span-4 lg:mt-2"
    role="row"
  >
    <span
      role="cell"
    >
      <a
        aria-label="React"
        href="https://react.dev/"
        rel="noopener noreferrer"
        tabindex="-1"
        target="_blank"
      >
        <img
          alt="logo of react"
          class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </a>
    </span>
    <span
      role="cell"
    >
      <a
        aria-label="Nextjs"
        href="https://nextjs.org/"
        rel="noopener noreferrer"
        tabindex="-1"
        target="_blank"
      >
        <img
          alt="logo of nextjs"
          class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </a>
    </span>
    <span
      role="cell"
    >
      <a
        aria-label="Spring Boot"
        href="https://spring.io/projects/spring-boot"
        rel="noopener noreferrer"
        tabindex="-1"
        target="_blank"
      >
        <img
          alt="logo of Spring Boot"
          class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </a>
    </span>
    <span
      role="cell"
    >
      <a
        aria-label="Docker"
        href="https://www.docker.com"
        rel="noopener noreferrer"
        tabindex="-1"
        target="_blank"
      >
        <img
          alt="logo of Docker"
          class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </a>
    </span>
    <span
      role="cell"
    >
      <a
        aria-label="Tailwind"
        href="https://tailwindcss.com/"
        rel="noopener noreferrer"
        tabindex="-1"
        target="_blank"
      >
        <img
          alt="logo of tailwind"
          class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </a>
    </span>
    <span
      role="cell"
    >
      <a
        aria-label="Zod"
        href="https://zod.dev"
        rel="noopener noreferrer"
        tabindex="-1"
        target="_blank"
      >
        <img
          alt="logo of zod"
          class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </a>
    </span>
    <span
      role="cell"
    >
      <a
        aria-label="Java"
        href="https://www.java.com/en/download/help/whatis_java.html"
        rel="noopener noreferrer"
        tabindex="-1"
        target="_blank"
      >
        <img
          alt="logo of java"
          class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </a>
    </span>
  </div>
</div>
`;
